mindsdb/interfaces/chatbot/chatbot_task.py

```diff
@@ -60,32 +59,38 @@ def run(self, stop_event):
         elif polling == 'realtime':
             self.chat_pooling = RealtimePolling(self, chat_params)
             self.memory = DBMemory(self, chat_params)
+
+        elif polling == 'webhook':
+            self.chat_pooling = WebhookPolling(self, chat_params)
+            self.memory = DBMemory(self, chat_params)
+
         else:
             raise Exception(f"Not supported polling: {polling}")
 
-        if self.agent_id is not None:
-            self.bot_executor_cls = AgentExecutor
-        elif self.bot_params.get('modes') is None:
-            self.bot_executor_cls = BotExecutor
-        else:
-            self.bot_executor_cls = MultiModeBotExecutor
+        self.bot_params['bot_username'] = self.chat_handler.get_my_user_name()
+
+    def run(self, stop_event):
+
+        # TODO check deleted, raise errors
+        # TODO checks on delete predictor / project/ integration
 
         self.chat_pooling.run(stop_event)
 
-    def on_message(self, chat_memory, message: ChatBotMessage, table_name=None):
+    def on_message(self, chat_id, message: ChatBotMessage, table_name=None):
 
         try:
-            self._on_message(chat_memory, message, table_name)
+            self._on_message(chat_id, message, table_name)
         except (SystemExit, KeyboardInterrupt):
             raise
         except Exception:
             error = traceback.format_exc()
             logger.error(error)
             self.set_error(str(error))
 
-    def _on_message(self, chat_memory, message: ChatBotMessage, table_name=None):
+    def _on_message(self, chat_id, message: ChatBotMessage, table_name=None):
         # add question to history
         # TODO move it to realtime pooling
+        chat_memory = self.memory.get_chat(chat_id, table_name=table_name)
```

Comment:
I see that there some caching in `ChatMemory` class. 
Potentialy it can lead to slowdown of [MessageCountPolling](https://github.com/mindsdb/mindsdb/blob/6539e789b8d12737bac5f7854d58dfeb6af8def9/mindsdb/interfaces/chatbot/polling.py#L43) where ChatMemory is initialized and then passed to `on_message`

Maybe to add chat_id as potional parameter and do?
```
chat_memory is None:
    chat_memory = self.memory.get_chat(chat_id, ...)
```